packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -723,16 +714,67 @@ describe("Decentraland", () => {
             action.once("confirmation", confirmationListener);
             action.on("error", errorListener);
 
-            // Tech-debt:
-            // Listen to the contract event that error will be thrown:  Error: done() called multiple times
-            // See more: https://github.com/tasitlabs/TasitSDK/issues/366
-            //gnosisSafe.once("ExecutionFailed", onFailed);
+            action.send();
+          })();
+        });
+
+        it("should buy an estate - using contract events", done => {
+          (async () => {
+            // Global hooks don't run between same level hooks
+            await mineBlocks(provider, 1);
+
+            const { assetId, nftAddress, priceInWei } = estateForSale;
+
+            const buyerAddress = GNOSIS_SAFE_ADDRESS;
+            await checkAsset(estate, mana, estateForSale, buyerAddress);
+
+            await expectExactTokenBalances(estate, [GNOSIS_SAFE_ADDRESS], [0]);
+
+            // Gnosis Safe should execute an open order
+            // TODO: ephemeralWallet should broadcast this action
+            const contractAddress = marketplace.getAddress();
+            const contractABI = marketplace.getABI();
+            const functionName = "safeExecuteOrder";
+            const fingerprint = await estate.getFingerprint(`${assetId}`);
+            const argsArray = [
+              nftAddress,
+              `${assetId}`,
+              `${priceInWei}`,
+              `${fingerprint}`,
+            ];
+
+            gnosisSafe.setSigners([gnosisSafeOwner]);
+            gnosisSafe.setWallet(gnosisSafeOwner);
+            const action = gnosisSafe.customContractAction(
+              contractAddress,
+              contractABI,
+              functionName,
+              argsArray
+            );
+
+            const onFailed = () => {
+              // WIP: Not working because of gas issue on Marketplace.safeExecuteOrder() call
+              //
+              // const { data } = message;
+              // const { args } = data;
+              // const { txHash } = args;
+              //done(new Error(`ExecutionFailed event emitted`));
+              done();
+            };
+
+            const errorListener = error => {
+              action.off("error");
+              done(error);
```

Comment:
Maybe note here that because ExecutionFailed is a custom contract event, we don’t pass it through as an error event yet. Otherwise we’d risk failing the test here. We may eventually want to detect that specifically for the Safe and call it an error.

Reply:
I didn't get it. Could you help me to understand?

Reply:
I believe that with a done-styled test, if you call `done(error)` the test fails. Right now, there's no way the `errorListener` would be triggered, so there's no way that `done(error)` would be called making the test fail.

But the contract event here is called `ExecutionFailed`, so you could imagine adding some `GnosisSafe` custom code that catches that event and passes it through as an error so the that `errorListener` would be called here. If that happened faster than the event listener for `ExecutionFailed` was called, then this test would intermittently fail.

So it's not a problem now, but it could be in the future.

Reply:
Ohh, got it now.
Shouldn't the test case fit with the current implementation and be changed if the `GnosisSafe` implementation changed as you've said?

Reply:
Yep, that's a good argument! We can resolve this.

Reply:
Resolving this.